mautic/mautic
# Make cache and log dir configurable before install.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This adjusts the PR in https://github.com/mautic/mautic/pull/12300 with latest M5 changes. 

copied from https://github.com/mautic/mautic/pull/12300

> Currently the checkstep takes a very biased approach where logs and cache are placed. This allows for these configurations to be overridden in local parameters and be loaded in before installation time.
> 
> Without this change, installing Mautic is not possible without first setting these folders to the "biased" location and only afterwards they can be moved.



#### Steps to test this PR:

1. Create a new mautic project using mautic/recommended-project
2. Set a file parameters_local.php in app/config before installing mautic with the following content
```
 <?php
 $parameters = [
   'cache_path' => '%kernel.project_dir%/../../var/cache',
   'log_path' => '%kernel.project_dir%/../../var/logs',
   'mailer_spool_path' => '%kernel.project_dir%/../../var/spool',
   'tmp_path' => '%kernel.project_dir%/../../var/tmp',
   'db_server_version' => '5.7.32',
. ];
```
3. Make sure you remove the docroot/var folder
4. Try installing mautic

Without this patch, installing mautic will fail because it thinks the folder is not writeable. In reality it actually is. 
With this patch the installation will be able to continue


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.
